src/error.rs

```diff
@@ -4,102 +4,57 @@
 // https://www.apache.org/licenses/LICENSE-2.0>. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use std::fmt;
-use thiserror::Error;
-
 /// Result Alias with PyroscopeError
 pub type Result<T> = std::result::Result<T, PyroscopeError>;
 
 /// Error type of Pyroscope
-#[derive(Error, Debug)]
-pub struct PyroscopeError {
-    pub msg: String,
-    source: Option<Box<dyn std::error::Error + Send + Sync>>,
-}
+#[derive(thiserror::Error, Debug)]
+pub enum PyroscopeError {
```

Comment:
making the error a pub enum
1. Is a breaking change
2. Means adding a variant is a breaking change (unless you use `#[non_exhaustive]`)
3. Don't really see the reason to be able to match on this error? It will be printed in any case. So it would be better to embed it as a private field into a public struct like before.

Reply:
Yeah, I'm not at all against these changes. With `#[non_exhaustive]` it might be future-proof enough.

Reply:
@drahnr @ordian How does this affect conversion later? The goal is to make it easy/possible to convert this type to the consumer of the library. Also does this automatically implement Backtrace? (at least on Nightly). I have a similar implementation here with Nightly Backtrace (https://github.com/rust-starter/rust-starter/blob/fbb5894b0419d99d83fb763402663e1bee5ad95e/utils/src/error.rs#L11). Would like your ideas/review on that.

Reply:
Added `#[non_exhaustive]`